datasource/src/test/scala/quasar/physical/mongo/DatasourceContractSpec.scala

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright 2014–2018 SlamData Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package quasar.physical.mongo
+
+import slamdata.Predef._
+import quasar.api.resource.{ResourceName, ResourcePath}
+import quasar.connector.{Datasource, DatasourceSpec, QueryResult}
+import quasar.physical.mongo.MongoDataSourceModule.ErrorOrResult
+import quasar.qscript.InterpretedRead
+
+import scala.io.Source
+
+import argonaut.Argonaut.jString
+import argonaut.Json
+import cats.effect.IO
+import fs2.Stream
+import testImplicits._
+
+class DatasourceContractSpec extends DatasourceSpec[IO, Stream[IO, ?]] {
+
+  val host = Source.fromResource("mongo-host").mkString.trim
+  val port: String = "27018"
+
+  val cfg = Json.obj(
+    "connectionString" -> jString(s"mongodb://root:secret@${host}:${port}"),
+    "pushdownLevel" -> jString("full"))
+
+  def ds: IO[ErrorOrResult[IO]] = MongoDataSourceModule.lightweightDatasource(cfg)
+
+  override val datasource: Datasource[IO, Stream[IO, ?], InterpretedRead[ResourcePath], QueryResult[IO]] =
```

Comment:
I think we should change `DatasourceSpec` in `quasar` so that we don't need to call unsafe code here. I have created https://app.clubhouse.io/data/story/5654/change-datasourcespec-so-that-it-better-integrates-with-datasource-implementations for this